src/components/CodeAndPreview.tsx

```diff
@@ -64,6 +64,13 @@ export function CodeAndPreview({ snippet }: Props) {
 
   const { toast } = useToast()
 
+  useEffect(() => {
+    if (snippet?.manual_edits_json) {
+      console.log('Setting manual edits from snippet:', snippet.manual_edits_json);
+      setManualEditsFileContent(snippet.manual_edits_json);
+    }
+  }, [snippet?.manual_edits_json])
```

Comment:
you only want to do this the first time it loads

```suggestion
  }, [Boolean(snippet?.manual_edits_json)])
```